codec/src/main/java/io/netty/handler/codec/LengthFieldBasedFrameDecoder.java

```diff
@@ -319,7 +316,7 @@ public LengthFieldBasedFrameDecoder(
                     "lengthFieldLength (" + lengthFieldLength + ").");
         }
 
-        this.byteOrder = byteOrder;
+        this.byteOrder = checkNotNull(byteOrder, "byteOrder");
```

Comment:
move the check and assignment before the other checks to ensure the behaviour is not changed.